bigquery_etl/cli/stage.py

```diff
@@ -203,15 +203,12 @@ def _udf_dependencies(artifact_files):
     for udf_file in udf_files:
         # all referenced UDFs need to be deployed in the same stage project due to access restrictions
         raw_routine = RawRoutine.from_file(udf_file)
-        udfs_to_publish = accumulate_dependencies([], raw_routines, raw_routine.name)
-
-        for dependency in udfs_to_publish:
-            dataset, name = dependency.split(".")
-            file_path = (
-                raw_routine.filepath.parent.parent.parent / dataset / name / UDF_FILE
-            )
```

Comment:
> @scholtzan any ideas on [this `deploy-changes-to-stage` error](https://app.circleci.com/pipelines/github/mozilla/bigquery-etl/25207/workflows/441baa1c-236f-42f8-a139-45ed34754d2e/jobs/262393)?
> 
> Seems like it's expecting some UDFs from `mozfun` to be in `moz-fx-data-shared-prod` instead for some reason.

I believe this was the cause of the `deploy-changes-to-stage` error (assuming the dependent UDFs were in the same project directory as the UDF calling them).

In hindsight I think I could have simply replaced this with logic using `raw_routines[dependency].filepath` without refactoring the `parse_routine` module so heavily to use fully qualified routine IDs, so if you think that refactoring is a bridge too far then I can try reverting it and just making this change instead.

Reply:
Never mind, I've gone ahead and reverted the refactoring of `parse_routine`.

A sticking point turned out to be that when ETLs are deployed to stage they're moved into `sql/bigquery-etl-integration-test/` and we lose any notion of which project they were in originally, so for any routine calls in the ETL that aren't fully qualified we can't tell for certain what their fully qualified routine IDs would be.

One possible solution for that would be to also deploy ETLs' UDF dependencies to stage like is done for views, then the partially qualified UDF calls would be resolvable to the staged UDFs, but that's really only necessary for ETLs with tests and I felt trying to do that was going to be too much for this PR, which was already overreaching its scope quite a bit.

I've reverted the refactoring I attempted rather than rebasing/erasing them just in case it might be useful for reference in some other future refactoring attempt.